src/eligibility-check-opkit.ts

```diff
@@ -0,0 +1,315 @@
+import { BotEvent, createReference, MedplumClient } from '@medplum/core';
+import {
+  Coverage,
+  CoverageEligibilityRequest,
+  CoverageEligibilityResponse,
+  Organization,
+  Patient,
+  Practitioner,
+  Reference,
+} from '@medplum/fhirtypes';
+import fetch from 'node-fetch';
+
+export async function handler(medplum: MedplumClient, event: BotEvent): Promise<any> {
+  // Because this bot is triggered by a subscription, the resource that comes in is a Coverage object
+  const coverage = event.input as Coverage;
+  const patient = await medplum.readReference(coverage.subscriber as Reference<Patient>);
+  const organization: Organization = await medplum.readReference(coverage.payor?.[0] as Reference<Organization>);
+  const provider: Practitioner = await medplum.readReference(
+    patient.generalPractitioner?.[0] as Reference<Practitioner>
+  );
+
+  if (!coverage) {
+    console.log('No coverage found');
+    return true;
+  }
+
+  if (!patient) {
+    console.log('No patient found');
+    return true;
+  }
+
+  if (!organization) {
+    console.log('No payor found');
+    return true;
+  }
+
+  if (!provider) {
+    console.log('No provider found');
+    return true;
+  }
+
+  var coverageEligibilityReq: CoverageEligibilityRequest = await medplum.createResource({
+    resourceType: 'CoverageEligibilityRequest',
+    provider: createReference(provider),
+    patient: createReference(patient),
+    insurer: createReference(organization),
+    insurance: [createReference(coverage)],
+  });
+
+  const providerNpi = provider.identifier?.find(
+    (identifier) => identifier.system === 'http://hl7.org/fhir/sid/us-npi'
+  )?.value;
+  const serviceTypes = ['health_benefit_plan_coverage'];
+  const opkitKey = '<opkit-public-api-key>';
+  const payerId = organization.identifier?.find(
+    (identifier) => identifier.system === 'https://docs.opkit.co/reference/getpayers'
+  )?.value;
+  const patientEmail = patient.telecom?.find((identifier) => identifier.system === 'email')?.value;
+
+  const opkitRequest = {
+    provider_npi: providerNpi,
+    payer_id: payerId,
+    subscriber: {
+      first_name: patient.name?.[0]?.given?.[0],
+      last_name: patient.name?.[0]?.family,
+      member_id: coverage.subscriberId,
+      date_of_birth: patient.birthDate,
+      email: patientEmail,
+    },
+    services: serviceTypes,
+  };
+
+  const result: any = await fetch('https://api.opkit.co/v1/eligibility_inquiries', {
```

Comment:
@codyebberson - what do you think about having a convenience wrapper around `medplum.post` to call out to apis. Thinking it could: 
- accept a mime type, but default to JSON
- do the JSON wrapping / unwrapping
- accept a param for {authType, authKey}